cmd/juju/cloud/add.go

```diff
@@ -139,6 +139,10 @@ func (c *AddCloudCommand) Run(ctxt *cmd.Context) error {
 		return errors.Errorf("cloud %q not found in file %q", c.Cloud, c.CloudFile)
 	}
 
+	// validate cloud data
+	if err := cloud.VerifyAuthTypes(newCloud.AuthTypes); err != nil {
```

Comment:
rather than defining a global list of auth-types, I think we should validate against the credential schemas defined by the provider?